src/randoop/main/GenTests.java

```diff
@@ -429,264 +399,335 @@ public boolean handle(String[] args) throws RandoopTextuiException {
     }
 
-
-    if (output_components != null) {
+    if (GenInputsAbstract.output_components != null) {
 
       assert explorer instanceof ForwardGenerator;
       ForwardGenerator gen = (ForwardGenerator)explorer;
 
-      // Output component sequences.
-      System.out.print("Serializing component sequences...");
-      try {
-        FileOutputStream fileos = new FileOutputStream(output_components);
-        ObjectOutputStream objectos = new ObjectOutputStream(new GZIPOutputStream(fileos));
-        Set<Sequence> componentset = gen.componentManager.getAllGeneratedSequences();
-        System.out.println(" (" + componentset.size() + " components) ");
-        objectos.writeObject(componentset);
-        objectos.close();
-        fileos.close();
-      } catch (Exception e) {
-        throw new Error(e);
-      }
+      outputComponentSequences(gen);
     }
 
-    if (dont_output_tests)
+    if (GenInputsAbstract.dont_output_tests)
       return true;
-
-    // Create JUnit files containing faults.
-    if (!GenInputsAbstract.noprogressdisplay) {
-      System.out.println();
-      System.out.print("Creating Junit tests (" + explorer.outSeqs.size() + " tests)...");
-    }
-    List<ExecutableSequence> sequences = new ArrayList<ExecutableSequence>();
-    for (ExecutableSequence p : explorer.outSeqs) {
-      sequences.add(p);
-    }
-
-    if (output_tests_serialized != null) {
-      // Output executable sequences.
-      System.out.println("Serializing tests...");
-      try {
-
-        FileOutputStream fileos = new FileOutputStream(output_tests_serialized);
-        ObjectOutputStream objectos = new ObjectOutputStream(new GZIPOutputStream(fileos));
-        objectos.writeObject(sequences);
-        objectos.close();
-        fileos.close();
-
-        FileInputStream fileis = new FileInputStream(output_tests_serialized);
-        ObjectInputStream objectis = new ObjectInputStream(new GZIPInputStream(fileis));
-        List<ExecutableSequence> seqsfromfile = (List<ExecutableSequence>) objectis.readObject();
-        assert seqsfromfile.size() == sequences.size();
-        for (int i = 0 ; i < seqsfromfile.size() ; i++) {
-          assert seqsfromfile.get(i).equals(sequences.get(i)) : seqsfromfile.get(i) + "@@@" + sequences.get(i);
-          ExecutableSequence eseq = seqsfromfile.get(i);
-          ReplayVisitor visitor = new ReplayVisitor();
-          eseq.execute(visitor);
-          assert eseq.getChecksResults().equals(sequences.get(i).getChecksResults()) :
-            sequences.get(i).getChecksResults() + "\n"
-            + seqsfromfile.get(i) + "\n"
-            + eseq.getChecksResults() + "\n" + sequences.get(i);
-
+  
+    if (! GenInputsAbstract.no_error_revealing_tests) {
+      List<ExecutableSequence> errorSequences = explorer.getErrorTestSequences();
+      if (errorSequences.size() > 0) {
+        if (! GenInputsAbstract.noprogressdisplay) {
+          System.out.format("%nError-revealing test output:%n");
```

Comment:
System.out.format and System.out.printf are the same, I believe.  Randoop uses the latter more than the former; I would continue to use printf, for consistency.